### PR TITLE
DOCSP-45930 Atlas Search Query Prep

### PR DESCRIPTION
**Pull request by ajhuh-mdb**
## DESCRIPTION
- Edit Compass Atlas Search Query page to:
  - Have more neutral wording that works for both Data Explorer & Compass 
  - Add Before You Begin section
  - Clean up procedures.

Other fixes to resolve some final build errors in https://github.com/10gen/cloud-docs/pull/6554:
- Updated some copy & screenshots on schema page, so they are easily reproducible for Data Explorer and reflect current terminology (e.g. `number` is referred to `int32`)
- Changed a manual link to the transactions page since it seems to conflict with another Atlas page.
- Marked the `export-agg-builder-results` page as Compass-only

## STAGING
https://deploy-preview-705--docs-compass.netlify.app/query/atlas-search/

## JIRA
https://jira.mongodb.org/browse/DOCSP-45930

## Self-Review Checklist

- [ ] Is this free of any warnings or errors in the RST?
- [ ] Is this free of spelling errors?
- [ ] Is this free of grammatical errors?
- [ ] Is this free of staging / rendering issues?
- [ ] Are all the links working?

## External Review Requirements

[What's expected of an external reviewer?](https://wiki.corp.mongodb.com/display/DE/Reviewing+Guidelines+for+the+MongoDB+Server+Documentation)

### CODE REVIEW

File: source/indexes.txt [hunk 1]
```diff
@@ -97,8 +95,6 @@ Atlas Vector Search indexes enable you to index vector data *and* other data
 types, which facilitates semantic searches on the indexed fields. For more
 information, see :ref:`compass-create-vector-search`.
```
Comment (ajhuh-mdb) [inline source/indexes.txt:96]: this ref and the ref above will link to the corresponding Atlas Search & Vector Search Index creation pages.

File: source/indexes.txt [hunk 2]
```diff
@@ -81,8 +81,6 @@ Create an Index
 
 .. include:: /includes/shared/steps/create-index.rst
 
-.. BEGIN-COMPASS-ONLY
-
 Create an Atlas Search Index
```
Comment (ajhuh-mdb) [inline source/indexes.txt:84]: Per Betsy's request, we're keeping this information here, so users on Data Explorer are directed to the correct place to create Search/Vector Search indexes.

File: source/schema.txt [hunk 3]
```diff
@@ -109,8 +109,8 @@ For fields that contain multiple data types,
 |compass-short| displays a percentage breakdown of
 the various data types across documents. In the example below,
 the chart shows the contents of a field called ``phone_no`` in which
-81% of documents are of type ``string``, and the remaining 19% are of
-type ``number``.
+20% of documents are of type ``int32``, and the remaining 80% are of
```
Comment (ajhuh-mdb) [inline source/schema.txt:112]: Changed this since the updated UI makes the percentage cover the collection name. Also, the ``number`` type doesn't exist anymore — it's now ``int32`` (or ``int64``)

File: source/query/atlas-search.txt [hunk 4]
```diff
@@ -12,11 +12,23 @@ Run Atlas Search Queries
...
+Before You Begin 
+----------------
+
+Create an :ref:`Atlas Search index <compass-create-search-index>` or an 
+:ref:`Atlas Vector Search index <compass-create-vector-search>`.
+
+.. note:: 
+   
+   You can manage Atlas Search indexes in the |compass-short| UI if your 
+   deployment is local, has an Atlas cluster tier of M10 or larger, and runs
```
Comment (sarah-olson-mongodb) [inline source/query/atlas-search.txt:30]: (minor suggestion)
"... has an Atlas M10 cluster tier or larger ..." 

(another minor suggestion)
This info might be more scannable in a list?

File: source/query/atlas-search.txt [hunk 5]
```diff
@@ -25,32 +37,28 @@ Steps
...
+      ``$searchMeta`` stage to your aggregation pipeline. If you have an 
+      Atlas Vector Search index, add the ``$vectorSearch`` stage.
 
-         .. note:: 
+      .. note:: 
 
-            The Atlas Search :pipeline:`$search` and :pipeline:`$searchMeta` and 
-            the Atlas Vector Search :pipeline:`$vectorSearch` pipeline stages must 
-            be the first stage in your aggregation pipeline.
+         The Atlas Search ``$search`` and ``$searchMeta`` and
```
Comment (sarah-olson-mongodb) [inline source/query/atlas-search.txt:57]: (question)
Is it EITHER the Atlas Vector Search pipeline stages, OR the Atlas Search pipeline stages? I'm not sure this makes sense as written, just yet.

File: source/query/atlas-search.txt [hunk 6]
```diff
@@ -12,11 +12,23 @@ Run Atlas Search Queries
    :depth: 1
    :class: singlecol
 
-You can run Atlas Search :pipeline:`$search` and :pipeline:`$searchMeta`
-queries from |compass| if you :ref:`created the Atlas Search index
-<compass-create-search-index>` for data on the Atlas cluster. You can
-also run :pipeline:`$vectorSearch` queries from |compass| if you have an
-existing Atlas Vector search index on your data in Atlas. 
+Atlas Search queries take the form of an :ref:`aggregation pipeline stage
```
Comment (sarah-olson-mongodb) [inline source/query/atlas-search.txt:15]: (minor)
"are run as" rather than "take the form of" 
Optional thought.

**Review by sarah-olson-mongodb — APPROVED** [review]
Thanks so much @ajhuh-mdb. These LGTM with a couple minor suggestions!

File: source/query/atlas-search.txt (hunk 6 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:15]: that sounds better. updated!

File: source/query/atlas-search.txt (hunk 5 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:57]: Ah let me try rewording this note altogether.

File: source/query/atlas-search.txt [hunk 7]
```diff
@@ -12,11 +12,26 @@ Run Atlas Search Queries
...
+Create an :ref:`Atlas Search index <compass-create-search-index>` or an 
+:ref:`Atlas Vector Search index <compass-create-vector-search>`.
+
+.. note:: 
+   
+   Atlas Search index management in the |compass-short| UI is available only 
+   for: 
+   
+   - Atlas local deployments.
+   - M10+ clusters.
```
Comment (betsybutton) [inline source/query/atlas-search.txt:33]: This restriction actually doesn't exist anymore -- you can manage search indexes on free/shared tier clusters in Compass. Would you mind removing this restriction as a drive-by fix in this PR?

File: source/query/atlas-search.txt [hunk 8]
```diff
@@ -25,32 +40,27 @@ Steps
...
-         By default, |compass| selects the :pipeline:`$search` stage for
-         your query. You can select the  :pipeline:`$searchMeta` stage
-         to use in your query. If you are using a ``vectorSearch`` type
-         index in your query, you can only use the
-         :pipeline:`$vectorSearch` stage in your query.  
+   .. step:: Construct your query
+      
+      If you have an Atlas Search index, add the ``$search`` or 
+      ``$searchMeta`` stage to your aggregation pipeline. If you have an 
+      Atlas Vector Search index, add the ``$vectorSearch`` stage.
```
Comment (betsybutton) [inline source/query/atlas-search.txt:56]: We actually do automatically create the $search or $vectorSearch stage for users if they click the "Aggregate" button from the search indexes list in Compass. Are we modifying this language to make it more compatible with DE? Want to make sure I understand why this change is happening

**Review by betsybutton — COMMENTED** [review]
Thank you Ali! Just a few comments

File: source/query/atlas-search.txt (hunk 7 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:33]: Can do!

File: source/query/atlas-search.txt (hunk 8 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:56]: Ah this might've stemmed from some testing I was doing on `cloud-dev` where it didn't automatically add the stage. Is this going to be the case where on DE, they have to add the search stage(s) themselves or will it have the same behavior as Compass?

File: source/query/atlas-search.txt (hunk 7 above)
Comment (betsybutton) [reply source/query/atlas-search.txt:33]: You won't be able to manage Atlas Search indexes in Data Explorer

File: source/query/atlas-search.txt (hunk 8 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:56]: EDIT: Given that this search index management won't be available in DE, it may make more sense to not migrate this page into the DE docs and instead link users to the [Atlas Search documentation](https://www.mongodb.com/docs/atlas/atlas-search/tutorial/run-query/) itself.

Comment (betsybutton) [reply source/query/atlas-search.txt:56]: Thanks Ali!

File: source/indexes.txt (hunk 2 above)
Comment (ajhuh-mdb) [reply source/indexes.txt:84]: Create a Search Index Page for Atlas: https://www.mongodb.com/docs/atlas/atlas-search/tutorial/create-index/

Comment (ajhuh-mdb) [reply source/indexes.txt:84]: Note that these links will be swapped out in https://github.com/10gen/cloud-docs/pull/6554

File: source/indexes.txt (hunk 1 above)
Comment (ajhuh-mdb) [reply source/indexes.txt:96]: Note that this link will be swapped out in https://github.com/10gen/cloud-docs/pull/6554 since the change will only pertain to Data Explorer

File: source/query/atlas-search.txt [hunk 9]
```diff
@@ -25,32 +39,28 @@ Steps
...
-         your query. You can select the  :pipeline:`$searchMeta` stage
-         to use in your query. If you are using a ``vectorSearch`` type
-         index in your query, you can only use the
-         :pipeline:`$vectorSearch` stage in your query.  
+   .. step:: Construct your query
+      
+      By default, |compass| selects the :pipeline:`$search` stage for your 
+      query. You can select the  :pipeline:`$searchMeta` stage to use in your 
+      query. If you are using a ``vectorSearch`` type index in your query, you 
+      can only use the :pipeline:`$vectorSearch` stage in your query.
```
Comment (betsybutton) [inline source/query/atlas-search.txt:56]: I know this was in the docs previously, but this actually isn't true. You can include other stages in the agg pipeline alongside $vectorSearch. Would it be convenient to make this change while we're here, or should I file a followup ticket?

File: source/query/atlas-search.txt [hunk 10]
```diff
@@ -25,32 +39,28 @@ Steps
...
+      can only use the :pipeline:`$vectorSearch` stage in your query. 
 
-         .. note:: 
+      .. note:: 
 
-            The Atlas Search :pipeline:`$search` and :pipeline:`$searchMeta` and 
-            the Atlas Vector Search :pipeline:`$vectorSearch` pipeline stages must 
-            be the first stage in your aggregation pipeline.
+         If you use the ``$search``, ``$searchMeta``, or ``$vectorSearch`` 
+         stage, it must be the first stage in your aggregation pipeline.
```
Comment (betsybutton) [inline source/query/atlas-search.txt:61]: This is mostly true, but hybrid search is an exception when both vector and full-text search are used in the same pipeline: https://www.mongodb.com/docs/atlas/atlas-search/tutorial/hybrid-search/

Is this worth calling out here?

File: source/query/atlas-search.txt (hunk 9 above)
Comment (ajhuh-mdb) [reply source/query/atlas-search.txt:56]: Will make this change here!

**Review by betsybutton — APPROVED** [review]
LGTM!

File: source/query/atlas-search.txt (hunk 10 above)
Comment (betsybutton) [reply source/query/atlas-search.txt:61]: Filed a followup: https://jira.mongodb.org/browse/DOCSP-46151
